src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.impl.BalanceHandler;
+import core.basesyntax.strategy.handler.impl.PurchaseHandler;
+import core.basesyntax.strategy.handler.impl.ReturnHandler;
+import core.basesyntax.strategy.handler.impl.SupplyHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlerMap
+            = Map.of(FruitTransaction.Operation.BALANCE, new BalanceHandler(),
+            FruitTransaction.Operation.SUPPLY, new SupplyHandler(),
+            FruitTransaction.Operation.PURCHASE, new PurchaseHandler(),
+            FruitTransaction.Operation.RETURN, new ReturnHandler());
```

Comment:
I'd recommend creating and initializing this map somewhere outside and pass it to the constructor of this class.